api/src/main/java/com/datastrato/gravitino/rel/types/Type.java

```diff
@@ -78,8 +78,16 @@ enum Name {
     /** The null type. A null type represents a value that is null. */
     NULL,
 
-    /** The unparsed type. An unparsed type represents an unresolvable type. */
-    UNPARSED
+    /**
+     * The unparsed type. An unparsed type represents an unresolvable type.
+     *
+     * @deprecated This type will be removed, use {@link #EXTERNAL} instead.
+     */
+    @Deprecated
+    UNPARSED,
```

Comment:
Does this `unparsed type` still work, will Gravitino still return this type for an unknown type?

Reply:
What happened if the client is a old client doesn't support external type, but server supports new external type, how to handle this compatibility issue?

Reply:
> Does this `unparsed type` still work, will Gravitino still return this type for an unknown type?

For current implementation, it will no longer work. it will return externalType instead of unparedType for  an unknown type

Reply:
> What happened if the client is a old client doesn't support external type, but server supports new external type, how to handle this compatibility issue?

#3599 has resolved the issue, it will be parsed as `unparsedType`

> Does this unparsed type still work, will Gravitino still return this type for an unknown type?

PR updated, unparsed type still works